src/service.rs

```diff
@@ -954,6 +954,50 @@ impl PowerEventParam {
     }
 }
 
+/// Enum describing the type of DeviceEvent event.
+///
+/// In order to receive those events, `RegisterDeviceNotification` must be called first.
+/// See <https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerdevicenotificationw>.
+///
+/// The value of each variant corresponds to the event data, which is kept in raw form as it can
+/// be quite complex. Please refer to the `lParam` documentation here for more details on this data:
+/// <https://learn.microsoft.com/en-us/windows/win32/devio/wm-devicechange>.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[non_exhaustive]
+pub enum DeviceEventParam {
```

Comment:
Given that each variant contains `*const c_void`, would it make sense to represent that as a struct with two fields instead?

Reply:
That would be quite equivalent yes, except that it would remove the possibility to add safe wrappers for some of those events in the future